Deploy to Dokku

adelq opened the pull request:
These commits _definitely_ need to be squashed, but leaving them for now to show progress.

adelq committed:
Update .gitignore with all Node.js ignores and remove platform-spec

All platform/editor specific edits should be in global gitignore

Give MacOS an exception in gitignore

Because global gitignores and code reviews are too hard :P

Ignore compiled assets in public/

Remove bundles from git

adelq committed:
Update package.json

adelq committed:
Add deploy-specific scripts to npm scripts

Remove rimraf dependency for production deploys

adelq committed:
Move dependencies around for production builds

Move needed webpack plugin to prod

Move babel dependencies into production

Move Procfile to root directory

Move babel loader into prod

Add style loader to build requirements

Move everything else needed to production

Reorder dependencies

ajmichaels commented on package.json:
```diff
@@ -1,59 +1,62 @@
+    "babel-polyfill": "^6.9.1",
+    "babel-preset-es2015": "^6.9.0",
     "babel-preset-stage-1": "^6.5.0",
+    "babel-preset-react": "^6.11.1",
```
Should be above stage-1

adelq replied:
Sharp eye, my manual alphabetization was not successful

ajmichaels commented on .gitignore:
```diff
@@ -1,9 +1,50 @@
-ApiServer/npm-debug.log
-.DS_Store
 data
+bundle.js
```
Rather than ignoring bundle.js, it's better to just ignore ApiServer/public, which contains bundle.js and style.css

adelq replied:
There's also a `bundle.js` generated at the root directory, so I'll leave that in but also add the public folder. Good call.

adelq replied:
Should I completely remove the `ApiServer/public` directory, or should I retain it with a `.gitkeep`?

ajmichaels replied:
Removing it is fine, since the build command will put it back anyway.

adelq commented on .gitignore:
```diff
@@ -1,9 +1,50 @@
+### App specific ###
 ApiServer/data
-ApiServer/npm-debug.log
-.DS_Store
```
Yes, some labs projects have .DS_Stores in them, please clean them up when you run into them, it's annoying when they slip in, that's what code reviews should be for. External contributors should not be a problem due to code reviews. Not to mention, MacOS isn't the only OS out there :wink:

adelq committed:
Add automatic deploys from Travis to Dokku

On a successful run of tests on the master branch, Travis CI will
automatically trigger the deploy script located in `deploy/deploy.sh`.

This script essentially loads the encrypted deploy private key into
ssh-agent and pushes the repo to dokku, which performs a deploy on
http://ohq.apps.pennlabs.org.

adelq commented:
@ajmichaels pull request is ready to go

ajmichaels commented:
@adelq Awesome, everything looks good.  Would you mind rebasing/merging off the latest master branch so we can resolve the conflicts?

adelq commented:
All done, squashed the commits into normal-sized chunks and merged into master. Let's see how this first deploy goes :)
